content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md

```diff
@@ -0,0 +1,267 @@
+
+However, if you have your scripts running outside the cluster that uses kubectl or a client library to access the Kubernetes cluster, you need the kubeconfig file to load the configs from. Follow the instructions in the next section to create such a kubeconfig file.
+
+
+
+### Create a Kubeconfig File
+Here is a shell script to create a kubeconfig file using the secret token of the service account. Update those variables in the script header to match with your environment.
+
+
+
+```bash
+# create-kubeconfig.sh
+
+# Update those variables to match your environment
+SERVICE_ACCOUNT_NAME="cfe-demo-sa"
+CONTEXT=$(kubectl config current-context)
+NEW_CONTEXT="cfe-demo-context"
+TOKEN_USER="cfe-token-user"
+KUBECONFIG_FILE="kubeconfig-sa"
+
+
+# Extract service account token
+SECRET_NAME=$(kubectl get serviceaccount ${SERVICE_ACCOUNT_NAME} --context ${CONTEXT}  -o jsonpath='{.secrets[0].name}')
+TOKEN_DATA=$(kubectl get secret ${SECRET_NAME}  --context ${CONTEXT}  -o jsonpath='{.data.token}')
+TOKEN=$(echo ${TOKEN_DATA} | base64 -d)
+
+#Create a general-purpose kubeconfig file
+kubectl config view --raw > tmp.raw
+kubectl --kubeconfig tmp.raw config use-context ${CONTEXT}
+kubectl --kubeconfig tmp.raw config view --flatten --minify > tmp.min
+kubectl config --kubeconfig tmp.min rename-context ${CONTEXT} ${NEW_CONTEXT}
+kubectl config --kubeconfig tmp.min  set-credentials ${TOKEN_USER} --token ${TOKEN}
+kubectl config --kubeconfig tmp.min set-context ${NEW_CONTEXT} --user ${ TOKEN_USER }
+kubectl config --kubeconfig tmp.min view --flatten --minify > ${KUBECONFIG_FILE}
+
+# Cleanup tmp
+rm tmp.raw
+rm tmp.min
+
+```
+After running the script, a general-purpose kubeconfig file `kubeconfig-sa` is created. By exporting the kubeconfig file as the environment variable `KUBECONFIG`, you can access the Kubernetes cluster and check all the resources deployed in the cluster.
+
+
+```bash
+$ bash create-kubeconfig.sh
+Switched to context "fab-zero-cfe-demo-cluster-cfe-demo-cluster-guoping.jia@hpe.com".
+Context "fab-zero-cfe-demo-cluster-cfe-demo-cluster-guoping.jia@hpe.com" renamed to "cfe-demo-context".
+User "cfe-token-user" set.
+Context "cfe-demo-context" modified.
+
+$ export KUBECONFIG=kubeconfig-sa
+
+$ kubectl get all
+No resources found in cfe-demo-cluster namespace.
+```
+## Conclusion
+This blog post shows you how to create a general-purpose kubeconfig file using a service account. This kubeconfig file is not tied to any specific user. It is bound with a list of permissions carefully chosen for your need to access the Kubernetes cluster. The created kubeconfig file works permanently with both the downloaded kubectl binary from HPE GreenLake for Containers dashboard and the standard one installed from the official Kubernetes site. This allows you to use it in any Kubernetes client scripts, esp., in your Kubernetes *CI/CD* pipeline.
+
```

Comment:
Also you can point to the user doc : https://support.hpe.com/hpesc/public/docDisplay?docId=a00092451en_us&page=HPE-GreenLake-for-containers.html